### DISCUSSION TITLE
C# Support?

### DISCUSSION THREAD

**Discussion by plumocracy** [utopia-rise/fmod-gdextension#240] [General]
Hi! My studio is hoping to use this plugin for our current project but we write mostly C# using the mono version of Godot. It seems theres no C# bindings for this plugin even though it says it can be used both with GDScript and C# in the readme. Is there C# support and I'm missing it somewhere or if not are there plans to add C# support soon? Thanks!

**Comment by CedNaru** [comment]
This is project is a GDExtension, which makes all its classes part of the Godot ClassDB. In theory, it means that any language can use this plugin as long as they provide a way to automatically bind to extensions. GDScript can easily do that, that's why it works out of the box whenever we write a script. I thought C# was able to automatically generate bindings for extensions as well, but it seems that feature is not yet available.
Unfortunately, we are not C# developers, so we can't make bindings specific to that language (on top of that it would be reimplementing everything a second time).

By the way, where do we mention that this binding can work with C# ?
If you are talking about this note:
`Note: FMOD also provides a C# wrapper for their API which is used in the Unity integration and it is possible to use the same wrapper to build an integration for Godot in C#. However do note that this would only work on a Mono build of Godot as C# support is required and performance might not be on the same level as a C++ integration.`
It doesn't mean we support C#, we simply mean that FMOD provide an API in C++ and C#, we simply picked the C++ one and glued it to the Godot API to make that plugin.

**Comment by CedNaru** [comment]
Maybe you can try plugins like this one (note that I don't know anything about it, it's just an attempt a making extensions available to C# while waiting for Godot to implement an official solution) :
https://github.com/Delsin-Yu/CSharp-Wrapper-Generator-for-GDExtension

**Comment by plumocracy** [comment]
Hey Ced! 

We've looked into plugins like Delsin's but because they're all still early in development they wont work for production. We've decided to go ahead and rewrite our codebase in GDScript for use of extensions because they're quite important for the quality we're trying to uphold. 

Also, I was referencing this line:
 
`This GDExtension provides nodes such as FmodEventEmitter2D, FmodEventEmitter3D, FmodEventListener2D and FmodEventListener3D that can be used in any Godot scene or code (both GDScript and C#). `

Which to me reads as if this extension is compatible with both supported languages. Appreciate you creating this plugin regardless! 

Here's a great issue regarding the GDExtension C# bindings if you're interested in following its progress. 

https://github.com/godotengine/godot-proposals/issues/8191

**Comment by CedNaru** [comment]
Our bad then. When we wrote this, we thought that C# already had an auto binding feature, but we assumed wrong. I fixed this statement.
